core/trino-main/src/main/java/io/trino/operator/Driver.java

```diff
@@ -654,20 +654,31 @@ private synchronized void checkLockHeld(String message)
         return result;
     }
 
-    // Note: task cannot return null
-    private <T> Optional<T> tryWithLock(Supplier<T> task)
+    /**
+     * Try to acquire the {@code exclusiveLock} immediately and run a {@code task}
+     * The task will not be interrupted if the {@code Driver} is closed.
+     * <p>
+     * Note: task cannot return null
+     */
+    private <T> Optional<T> tryWithLockUninterruptibly(Supplier<T> task)
     {
-        return tryWithLock(0, TimeUnit.MILLISECONDS, task);
+        return tryWithLock(0, TimeUnit.MILLISECONDS, false, task);
     }
 
-    // Note: task cannot return null
-    private <T> Optional<T> tryWithLock(long timeout, TimeUnit unit, Supplier<T> task)
```

Comment:
nit: keep this method maybe so that you don't have to add `true` argument for existing `tryWithLock` invodations

Reply:
Since this is a private method I wasn't sure if it is worth adding extra overloads (as then you have to document them as well :-))